code/modules/mob/living/carbon/alien/special/alien_embryo.dm

```diff
@@ -46,11 +54,16 @@
 			to_chat(owner, "<span class='danger'>You feel something tearing its way out of your stomach...</span>")
 			owner.adjustToxLoss(10)
 
-/obj/item/organ/body_egg/alien_embryo/egg_process()
-	if(stage < 5 && prob(3))
+/obj/item/organ/body_egg/alien_embryo/proc/advance_embryo_stage()
+	if(stage >= 5)
+		return
+	if(stage < 5)
 		stage++
 		INVOKE_ASYNC(src, .proc/RefreshInfectionImage)
+		addtimer(CALLBACK(src, .proc/advance_embryo_stage), growth_time)
+	return
```

Comment:
Don't need this return at the end of the function, since this is the end of the function anyway.